### PR TITLE
CopilotChat: Show uploaded documents in chat history

### PR DESCRIPTION
**Pull request by dehoward**
### Motivation and Context

Now when uploading a file in the chat a card will be shown indicating a success rather than the alert banner at the top.

### Description

- adds `type` to the `ChatMessage` model
- updates `ChatHistory` to render the new `ChatHistoryFileItem` or the existing `ChatHistoryItem` based on the `type` of the message
- updates `ChatSkill` to not generate a bot response if the message `type` is `ChatMessageType.Document`

### Preview
gif

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Models/ChatMessage.cs [hunk 1]
```diff
@@ -34,6 +34,27 @@ public enum AuthorRoles
...
+        /// </summary>
+        Message,
+
+        /// <summary>
+        /// A message for a Plan
+        /// </summary>
+        Plan,
+
+        /// <summary>
+        /// An uploaded document
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Models/ChatMessage.cs:53]:
```suggestion
        /// An uploaded document notification
```

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Models/MemorySource.cs [hunk 2]
```diff
@@ -63,14 +63,20 @@ public enum MemorySourceType
     [JsonPropertyName("createdOn")]
     public DateTimeOffset CreatedOn { get; set; }
 
+    /// <summary>
+    /// The size of the source in bytes.
+    /// </summary>
+    [JsonPropertyName("Size")]
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Models/MemorySource.cs:69]:
```suggestion
    [JsonPropertyName("size")]
```

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/apps/copilot-chat-app/webapp/src/libs/services/BaseService.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ interface ServiceRequest {
     body?: FormData | unknown;
 }
 const noResponseBodyStatusCodes = [202];
-const noResponseBodyOperations = ['importDocument'];
+const noResponseBodyOperations: string[] = [];
```
Comment (teresaqhoang) [inline samples/apps/copilot-chat-app/webapp/src/libs/services/BaseService.ts:11]: nit: we should just remove this if it's not used

File: samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts [hunk 4]
```diff
@@ -116,6 +126,10 @@ export const useChat = () => {
                     key: 'chatId',
                     value: chatId,
                 },
+                {
+                    key: 'type',
```
Comment (teresaqhoang) [inline samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts:130]: nit: let's rename this to a more descriptive `messageType`

File: samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts [hunk 5]
```diff
@@ -146,14 +160,22 @@ export const useChat = () => {
...
 
+            // When a document is uploaded we only save the user's message to storage, and
+            // do not generate a bot response. Therefore, we do not need to update the conversation.
+            if (type === ChatMessageType.Document) {
+                return;
+            }
+
+            const isPlan = PlanUtils.isPlan(result.value);
             const messageResult = {
+                type: isPlan ? ChatMessageType.Plan : ChatMessageType.Message,
```
Comment (teresaqhoang) [inline samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts:171]: MessageType should be returned either in the context or chat message result, right? Can we just assign that value here?

Comment (dehoward) [reply samples/apps/copilot-chat-app/webapp/src/libs/useChat.ts]: good point. this is now returned in the context.

**Review by teresaqhoang — APPROVED** [review]

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**